Analysis, Cleaning, and Construction Checklists for Peer Code Review (#43)
worldbank/dime-standards: dime-coding-standards/checklists/tex/cleaning_code_review_checklist.tex

\newcommand{\dimeTextField}[3]{\TextField[name=#1, height=0.3cm, width=#2, bordercolor=gray]{#3}}

\newcommand{\titleBox}[1]{
	\begin{tcolorbox}
		[colframe = fontcolor,
		colback = fontcolor,
		sharp corners,
		halign = flush center,
		valign = center,
		height = 0.3\textwidth,
		after skip = 1cm]
		#1
	\end{tcolorbox}
}

\begin{document}
	\begin{fullwidth}
		
		\titleBox{
			\textcolor{white}{\LARGE{\textbf{DIME Analytics \\ Peer Code Review - Cleaning Checklist}} \\
				\Large\textbf{{v1.0}}}
		}

  \dimeTextField{time}{2.5 cm}{Reviewer:} 

        \dimeTextField{time}{2.5 cm} {Coder:}
		
		\textbf{NOTE:} Make sure that fill out this checklist \textbf{ONLY IF} your partner's submission includes    \textbf{cleaning} tasks. 
		
		\section*{Data cleaning tasks}
  This checklist lists important factors to consider while reviewing your code review partner's \textbf{cleaning} scripts/code. Please fill this checklist, and submit it as an attachment when you submit \href{https://survey.wb.surveycto.com/collect/code_review_summary?caseid=}{this detailed form.}
		
		\begin{itemize}
			\setlength\itemsep{-0.1em}

> 			\item[] \dimeCheckBox{Are new variables being created in the cleaning do-files?}

Reviewer: ```suggestion
			\item[] \dimeCheckBox{Cleaning scripts do not create indicators (this should be done in a separate construction script)}
```

>               \item[] \dimeCheckBox{Are any changes being made to observations values in the cleaning do-files?}

Reviewer: ```suggestion
              \item[] \dimeCheckBox{Any changes made to specific values are well-documented and are justifiable. }
```

              \item[]\dimeCheckBox{\textbf{Check duplicates:}} 
            
            \begin{itemize}

>                 \item[] \dimeCheckBox{Does the code check for duplicates in ID variables and other important variables?}

Reviewer: ```suggestion
                \item[] \dimeCheckBox{The code checks for duplicates in ID variables and other important variables?}
```

>                 \item[] \dimeCheckBox{If so, are the duplicates fixed?}

Reviewer: ```suggestion
                \item[] \dimeCheckBox{All identified duplicates are resolved in code, or explanation is provided for why they cannot be resolved}
```

>                 \item[] \dimeCheckBox{In cases where duplicates are not resolved, is there a comment explaining next steps?}

Reviewer: ```suggestion
```

            \end{itemize}
			\item[] \dimeCheckBox{\textbf{Check merges:}} 
            
            \begin{itemize}

>                 \item[] \dimeCheckBox{Are any observations dropped?}

Reviewer: ```suggestion
                \item[] \dimeCheckBox{If any observations are dropped, there is a clear justification in the code}
```

>                 \item[] \dimeCheckBox{If so, is there a clear justification for that?}

Reviewer: ```suggestion
```

>                 \item[] \dimeCheckBox{If any observations didn't match, is that explained in the comments?}

Reviewer: ```suggestion
                \item[] \dimeCheckBox{Any mismatches between datasets are explained in the code}
```

            \end{itemize}
            \item[] \dimeCheckBox{\textbf{Check labels:}} 
            
            \begin{itemize}

>                 \item[] \dimeCheckBox{Are all variables correctly labelled?}

Reviewer: ```suggestion
                \item[] \dimeCheckBox{All variables are correctly labelled}
```

>                 \item[] \dimeCheckBox{If so, are the labels consistent?}

Reviewer: ```suggestion
                \item[] \dimeCheckBox{All value labels are consistent (example of inconsistency: varA 1 = yes, 0 = no; varB 1 = yes, 2 = no)}
```

            \end{itemize}

            \dimeCheckBox{\textbf{Check missing values:}} 
            \begin{itemize}

> 			\item[] \dimeCheckBox{Are missing values coded consistently?}

Reviewer: ```suggestion
			\item[] \dimeCheckBox{Missing values are coded consistently}
```

>                 \item[] \dimeCheckBox{Are extended missing values used? (e.g. \texttt{.d} for \textbf{Do not know}, \texttt{.r} for \textbf{Refuse to answer}, etc.}

Reviewer: ```suggestion
                \item[] \dimeCheckBox{Extended missing values are used (e.g., \texttt{.d} for \textbf{Do not know}, \texttt{.r} for \textbf{Refuse to answer}, etc.}
```
